File: solr-facade/src/main/java/com/cherrysoft/solrfacade/model/SearchResult.java
package com.cherrysoft.solrfacade.model;

import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public final class SearchResult {
  public static final SearchResult EMPTY = new SearchResult();
  private final List<RecoveredDocument> recoveredDocuments;
  private SpellcheckResult spellcheckResult;

  public SearchResult() {
    this.recoveredDocuments = new ArrayList<>();
  }

  public void addWebPageDocuments(List<RecoveredDocument> documents) {
    recoveredDocuments.addAll(documents);
  }

  @Getter
  @Builder
  @RequiredArgsConstructor
  public static class SpellcheckResult {
    private final List<String> suggestions;
    private final List<String> collations;
  }

}

File: frontend/src/models/search/documents/SearchResult.ts
import { SpellcheckResult } from '../spellcheck/SpellcheckResult';
import { RetrievedDocument } from './RetrievedDocument';

export interface SearchResult {
  totalDocsFound: number;
  documents: RetrievedDocument[];
  spellcheckResult: SpellcheckResult;
}
